src/main/lombok/org/heigit/ohsome/ohsomeapi/inputprocessing/GeometryBuilder.java

```diff
@@ -225,17 +225,22 @@ public void createGeometryFromMetadataGeoJson(String geoJson) {
   public Geometry createGeometryFromGeoJson(String geoJson, InputProcessor inputProcessor) {
     ArrayList<Geometry> geometryList = new ArrayList<>();
     JsonObject root = null;
+    JsonArray features;
+    Object[] boundaryIds;
+    GeoJsonObject[] geoJsonGeoms;
     try (JsonReader jsonReader = Json.createReader(new StringReader(geoJson))) {
       root = jsonReader.readObject();
+      features = root.getJsonArray("features");
+      boundaryIds = new Object[features.size()];
+      geoJsonGeoms = new GeoJsonObject[features.size()];
     } catch (Exception e) {
-      throw new BadRequestException("Error in reading of the given GeoJSON.");
+      throw new BadRequestException("Error in reading the provided GeoJSON. Please take a look at "
+          + "the documentation page for the bpolys parameter to see an example of a fitting "
+          + "GeoJSON input file.");
     }
     if (!"FeatureCollection".equals(root.getString("type"))) {
       throw new BadRequestException("The given GeoJSON has to be of the type 'FeatureCollection'.");
     }
-    JsonArray features = root.getJsonArray("features");
-    Object[] boundaryIds = new Object[features.size()];
-    GeoJsonObject[] geoJsonGeoms = new GeoJsonObject[features.size()];
```

Comment:
PS: another option would be to use a json schema validator like https://github.com/everit-org/json-schema or https://github.com/networknt/json-schema-validator to check if the user input matches the official GeoJSON schema of a feature collection: https://geojson.org/schema/FeatureCollection.json

Reply:
I'd prefer your first suggestion here, since it does not increase the code complexity and still gives the necessary info to the user.

Regarding the schema validator: https://github.com/networknt/json-schema-validator actually looks nice as it also uses Jackson. It would add some processing overhead though, so the question is if it would be necessary, or if the current implementation is enough.

Reply:
> I'd prefer your first suggestion here, since it does not increase the code complexity and still gives the necessary info to the user.

Seems reasonable to me as well :+1: We can keep the json validation option in mind for further improvements in the future.